src/object.d

```diff
@@ -3241,6 +3241,40 @@ template RTInfo(T)
     enum RTInfo = null;
 }
 
+bool __equals(T1, T2)(T1[] lhs, T2[] rhs)
+{
+    if (lhs.length != rhs.length)
+        return 0;
```

Comment:
stylistic nitpick: if you're going to use `return true;` at the end, then make this `return false;`